### PR TITLE
Merging r1.0 to master

### PR DESCRIPTION
**Pull request by av8ramit**
Merging r1.0 back into master to include all the commits that were needed for 1.0.1.

### CODE REVIEW

**Comment by jhseu** [comment]
This likely won't work, and none of the changes seem that important (most are only relevant to 1.0). Let's not merge back in?

**Comment by jhseu** [comment]
IMO, only the doc changes are useful. The others may not compile and conflict with what's in master. @caisq can comment on whether the `stepper_test` changes are useful.

File: tensorflow/core/graph/graph_constructor.cc [hunk 1]
```diff
@@ -821,12 +821,11 @@ Status GraphConstructor::MakeEdge(Node* src, int output_index, Node* dst,
 Status ConvertGraphDefToGraph(const GraphConstructorOptions& opts,
```
Comment (cwhipkey) [inline tensorflow/core/graph/graph_constructor.cc]: remove these two changes?  Seems like the version at master is better?

File: tensorflow/core/public/version.h [hunk 2]
```diff
@@ -20,7 +20,7 @@ limitations under the License.
 
 #define TF_MAJOR_VERSION 1
 #define TF_MINOR_VERSION 0
-#define TF_PATCH_VERSION 0
+#define TF_PATCH_VERSION 1
```
Comment (cwhipkey) [inline tensorflow/core/public/version.h]: do we expect to merge this?

File: tensorflow/tools/pip_package/setup.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
 # This version string is semver compatible, but incompatible with pip.
 # For pip, we will remove all '-' characters from this string, and use the
 # result for pip.
-_VERSION = '1.0.0'
+_VERSION = '1.0.1'
```
Comment (cwhipkey) [inline tensorflow/tools/pip_package/setup.py]: is this expected to be merged back?

File: tensorflow/tools/ci_build/Dockerfile.debian.jessie.cpu [hunk 4]
```diff
@@ -5,7 +5,15 @@ MAINTAINER Jan Prach <jendap@google.com>
 # Copy and run the install scripts.
 COPY install/*.sh /install/
```
Comment (cwhipkey) [inline tensorflow/tools/ci_build/Dockerfile.debian.jessie.cpu]: for this and the next ones, are these changes that were made to 1.0.1 and need backporting?

**Comment by cwhipkey** [comment]
+1 to what Jonathan said.

**Comment by gunan** [comment]
We have to merge a few changes:
-changing the ci_build Dockerfiles to use ubuntu 14.04
-fixes to mac_gpu tests.

We can pick master version for most changes, but we should merge the branch back into master.

File: tensorflow/tools/pip_package/setup.py (hunk 3 above)
Comment (av8ramit) [inline tensorflow/tools/pip_package/setup.py]: Yes we are merging this back in. cc: @gunan

File: tensorflow/tools/ci_build/ci_parameterized_build.sh [hunk 5]
```diff
@@ -314,6 +313,12 @@ else
     MAVX2_FMA_DBG)
       OPT_FLAG="${OPT_FLAG} -c opt --copt=-g --copt=-mavx2 --copt=-mfma"
       ;;
+    MAVX2_FMA)
```
Comment (jhseu) [inline tensorflow/tools/ci_build/ci_parameterized_build.sh]: These changes are duplicated. Delete?

File: tensorflow/tools/ci_build/Dockerfile.gpu [hunk 6]
```diff
@@ -1,7 +1,12 @@
-FROM nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04
+FROM nvidia/cuda:8.0-cudnn5-devel-ubuntu14.04
```
Comment (jhseu) [inline tensorflow/tools/ci_build/Dockerfile.gpu]: Note that if we merge this in, tests will break for two reasons:
- The tests in master no longer work with Ubuntu 14.04. We have to restore the python 3.5 stuff.
- The tests are known to crash on Ubuntu 14.04 unless we install numpy from scratch. We have to update that too.

Delete for now?

File: tensorflow/core/common_runtime/shape_refiner.cc [hunk 7]
```diff
@@ -437,4 +437,4 @@ Status ShapeRefiner::ConstantPartialShape(InferenceContext* target_context,
   return Status::OK();
 }
 
-}  // namespace tensorflow
+}  // namespace tensorflow
```
Comment (jhseu) [inline tensorflow/core/common_runtime/shape_refiner.cc]: Revert this too?

File: tensorflow/core/common_runtime/shape_refiner.h [hunk 8]
```diff
@@ -123,4 +123,4 @@ class ShapeRefiner {
 
 }  // namespace tensorflow
 
-#endif  // THIRD_PARTY_TENSORFLOW_CORE_COMMON_RUNTIME_SHAPE_REFINER_H_
+#endif  // THIRD_PARTY_TENSORFLOW_CORE_COMMON_RUNTIME_SHAPE_REFINER_H_
```
Comment (jhseu) [inline tensorflow/core/common_runtime/shape_refiner.h]: Revert?

File: tensorflow/core/graph/graph_constructor.cc [hunk 9]
```diff
@@ -878,4 +878,4 @@ void CopyGraph(const Graph& src, Graph* dest) {
   }
 }
 
-}  // namespace tensorflow
+}  // namespace tensorflow
```
Comment (jhseu) [inline tensorflow/core/graph/graph_constructor.cc]: Revert?

File: tensorflow/cc/framework/scope.cc [hunk 10]
```diff
@@ -464,4 +464,4 @@ CompositeOpScopes Scope::GetCompositeOpScopes(
   }
 }
 
-}  // namespace tensorflow
+}  // namespace tensorflow
```
Comment (jhseu) [inline tensorflow/cc/framework/scope.cc]: Revert missing newline?

File: tensorflow/tools/ci_build/Dockerfile.gpu (hunk 6 above)
Comment (gunan) [inline tensorflow/tools/ci_build/Dockerfile.gpu]: As I mentioned before, this governs the package we build for releases.
We cannot build 1.1 release on 16.04 -> so this has to be merged.
Could you point Amit to the changes that need to be done, so we can add them to this PR?

**Comment by av8ramit** [comment]
Sorry reverted to the wrong version of cwhipkey's changes, fixing now.

**Comment by av8ramit** [comment]
Messed up a few things, had some confusion on what should go in. I'll just create a new PR.

**Pull request closed without merging**